src/db/moderator.ts

```diff
@@ -0,0 +1,49 @@
+import { Snowflake } from 'discord.js';
+import { Schema } from 'mongoose';
+
+/**
+ * A moderator is similar to a user, except
+ * we **do not** keep track of previous usernames
+ * or nicknames.
+ *
+ * This isn't saved by itself, rather it is used to "log"
+ * actions taken by an admin.
+ */
+export interface Moderator {
+  /**
+   * The discord server userId for the mod
+   */
+  userId: string | Snowflake;
+  /**
+   * The discord server nickname of the mod
+   */
+  username: string;
+  /**
+   * The discord server nickname of the mod
+   */
+  nickname: string;
```

Comment:
To clarify, the nickname can be `null`. You and I both do not have nicknames set, so this field is empty. To typecast as a string, I used the `member.nickname` field and a fallback of `""`.

Reply:
Ah I didn't know that good call! I'll update the comments and schema to set this as an empty string as default.